python/housinginsights/ingestion/LoadData.py

```diff
@@ -286,8 +289,267 @@ def rebuild(self):
+        self._create_zone_facts_table()
+
         return processed_data_ids
 
+    def _create_zone_facts_table(self):
+        """
+        Creates the zone_facts table which is used as a master table for API
+        endpoints. The purpose is to avoid recalculating fields adhoc and
+        performing client-side reconfiguration of data into display fields.
+        This is all now done in the backend whenever new data is loaded.
+
+        :return: a immutable dict that contains the tables structure of the
+        table and sqlAlchemy metadata object used.
+        """
+        # drop zone_facts table if already in db
+        if 'zone_facts' in self.engine.table_names():
+            with self.engine.connect() as conn:
+                conn.execute('DROP TABLE zone_facts;')
+
+        # create empty zone_facts table
+        metadata = MetaData(bind=self.engine)
+        zone_facts = Table('zone_facts', metadata,
+                           Column('zone', String, primary_key=True),
+                           Column('poverty_rate', Numeric),
+                           Column('fraction_black', Numeric),
+                           Column('income_per_capita', Numeric),
+                           Column('labor_participation', Numeric),
+                           Column('fraction_foreign', Numeric),
+                           Column('fraction_single_mothers', Numeric),
+                           Column('acs_lower_rent_quartile', Numeric),
+                           Column('acs_median_rent', Numeric),
+                           Column('acs_upper_rent_quartile', Numeric))
+
+        # add to db
+        metadata.create_all(tables=[zone_facts])
+
+        # populate table with calculated fields values
+        self._populate_zone_facts_table()
+
+        return metadata.tables
+
+    def _populate_zone_facts_table(self):
+        """
+        Populates the zone_facts table with the calculated fields data and
+        acs rent data fields from census.
+
+        :return: list of SQLAlchemy query result objects from inserting the
+        calculated fields data into the zone_facts table.
+        """
+        census_fields = [
+            'poverty_rate', 'fraction_black', 'income_per_capita',
+            'labor_participation', 'fraction_foreign',
+            'fraction_single_mothers', 'acs_lower_rent_quartile',
+            'acs_median_rent', 'acs_upper_rent_quartile'
+        ]
+
+        zone_types = ['ward', 'neighborhood_cluster', 'census_tract']
+
+        query_results = list()
+
+        # populate columns accordingly for each zone_specific type
+        for zone_type in zone_types:
+            field_values = dict()
+
+            # get field value for each zone_specific type
+            for field in census_fields:
+                result = self._census_with_weighting(data_id=field,
+                                                     grouping=zone_type)
+                field_values[field] = result['items']
```

Comment:
In this context, returning the dictionary that includes the 'items' key probably doesn't make sense since we'll be writing this straight to the database, not making a user interface. I'd suggest refactoring _census_with_weighting to just return the content of 'items' and then this field can fust be field_values[field] = result